workspaces/util/src/__tests__/swap-test.ts

```diff
@@ -0,0 +1,66 @@
+import { describe, expect, it } from "@jest/globals";
+
+import { swap } from "../swap";
+
+describe("swap", () => {
+  it("swaps two elements in an array of numbers", () => {
+    const arr = [1, 2, 3, 4];
+    swap(arr, 1, 3);
+    expect(arr).toStrictEqual([1, 4, 3, 2]);
+  });
+
+  it("swaps two elements in an array of strings", () => {
+    const arr = ["a", "b", "c", "d"];
+    swap(arr, 0, 2);
+    expect(arr).toStrictEqual(["c", "b", "a", "d"]);
+  });
+
+  it("swaps two elements in an array of mixed types", () => {
+    const arr = [1, "a", true, 3.14];
```

Comment:
If you want to be fancy 😄 

```suggestion
    const arr = [1, "a", true, Math.PI];
```